s3-source-connector/src/integration-test/java/io/aiven/kafka/connect/s3/source/IntegrationTest.java

```diff
@@ -0,0 +1,401 @@
+        connectRunner.createConnector(connectorConfig);
+
+        final String testData1 = "Hello, Kafka Connect S3 Source! object 1";
+        final String testData2 = "Hello, Kafka Connect S3 Source! object 2";
+
+        // write 2 objects to s3
+        writeToS3(topicName, testData1.getBytes(StandardCharsets.UTF_8), "00000");
+        writeToS3(topicName, testData2.getBytes(StandardCharsets.UTF_8), "00000");
+        writeToS3(topicName, testData1.getBytes(StandardCharsets.UTF_8), "00001");
+        writeToS3(topicName, testData2.getBytes(StandardCharsets.UTF_8), "00001");
+        writeToS3(topicName, new byte[0], "00003"); // this should be ignored.
+
+        final List<String> objects = testBucketAccessor.listObjects();
+        assertThat(objects.size()).isEqualTo(5);
+
+        // Verify that the connector is correctly set up
+        assertThat(connectorConfig.get("name")).isEqualTo(CONNECTOR_NAME);
+
+        // Poll messages from the Kafka topic and verify the consumed data
+        final List<String> records = IntegrationBase.consumeMessages(topicName, 4, KAFKA_CONTAINER);
+
+        // Verify that the correct data is read from the S3 bucket and pushed to Kafka
+        assertThat(records).contains(testData1).contains(testData2);
+    }
+
+    @Test
+    void bytesTestBasedOnMaxMessageBytes(final TestInfo testInfo)
+            throws ExecutionException, InterruptedException, IOException {
+        final String testData = "AABBCCDDEE";
+        final var topicName = IntegrationBase.topicName(testInfo);
+        final Map<String, String> connectorConfig = getConfig(basicConnectorConfig(CONNECTOR_NAME), topicName);
+        connectorConfig.put(INPUT_FORMAT_KEY, InputFormat.BYTES.getValue());
+        connectorConfig.put(EXPECTED_MAX_MESSAGE_BYTES, "2"); // For above test data of 10 bytes length, with 2 bytes
+                                                              // each
+        // in source record, we expect 5 records.
+        connectorConfig.put(MAX_POLL_RECORDS, "2"); // In 3 polls all the 5 records should be processed
+
+        connectRunner.createConnector(connectorConfig);
+
+        writeToS3(topicName, testData.getBytes(StandardCharsets.UTF_8), "00000");
+
+        // Poll messages from the Kafka topic and verify the consumed data
+        final List<String> records = IntegrationBase.consumeMessages(topicName, 5, KAFKA_CONTAINER);
+
+        // Verify that the correct data is read from the S3 bucket and pushed to Kafka
+        assertThat(records.size()).isEqualTo(5);
+        assertThat(records.get(0)).isEqualTo("AA");
+        assertThat(records.get(1)).isEqualTo("BB");
+        assertThat(records.get(2)).isEqualTo("CC");
+        assertThat(records.get(3)).isEqualTo("DD");
+        assertThat(records.get(4)).isEqualTo("EE");
+    }
+
+    @Test
+    void multiPartUploadBytesTest(final TestInfo testInfo) throws ExecutionException, InterruptedException {
```

Comment:
I don't think this test is required, as it is testing the multipart upload rather then the source connector. 
The multipart upload writes the file on closing of the stream, so it appears as any old file to the source connector.